### PR TITLE
[HW_8][JS_6]

### PR DESCRIPTION
**Pull request by JackBlaaack**
work with objects

### CODE REVIEW

File: JS_6/task2.2.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+const getEnterpriseName = ((search) => {
+    for (const enterprise of enterprises) {
+        for (const department of enterprise.departments) {
+            if (department.id === search || department.name === search) {
+                return enterprise.name;
+            }
+        }
+    }
+    return `Предприятие не найдено`
```
Comment (anatoly-karpovich) [inline JS_6/task2.2.js:18]:
```suggestion
    throw new Error(`Предприятие не найдено`)
```
If function returns an object - you shall not return string in any case. Only throwing an Error is ok

File: JS_6/task2.3.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// 3. Написать функцию, которая будет добавлять предприятие. В качестве аргумента принимает название предприятия
+
+// Пример:
+// addEnterprise("Название нового предприятия")
+
+import { enterprises } from "./task2.1.js";
+
+const addEnterprise = (name => {
+
+    const newId = Math.max(...enterprises.map(enterprise => enterprise.id)) + 1;
```
Comment (anatoly-karpovich) [inline JS_6/task2.3.js:10]: Incorrect. New id must be found through all departments too. All ids must be unique among all objects.

And of cause this code should be moved to an helper function generateNewId

File: JS_6/task2.4.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import { enterprises } from "./task2.1.js";
+
+const addDepartments = (entId, departName) => {
+    try {
+        const enterprise = enterprises.find(ent => ent.id === entId); // находим предприятие по id
+        if (!enterprise) {
+            throw new Error('Такого предприятия нет');
+        }
+
+        const newId = Math.max(...enterprises.flatMap(ent => {
```
Comment (anatoly-karpovich) [inline JS_6/task2.4.js:14]: Same as above

File: JS_6/task2.9.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+                if (dep.id === fromDepId) {
+                    fromDep = dep;
+                if (dep.id === toDepId) {
+                    toDep = dep;
+                }
+                }
+            })
+        })
+        if (fromDep && toDep) {
+            toDep.employees_count += fromDep.employees_count;
```
Comment (anatoly-karpovich) [inline JS_6/task2.9.js:22]: No check that departments are part of one enterprise

**Review by anatoly-karpovich — DISMISSED** [review]

**Review by VladimirLQA — DISMISSED** [review]

File: JS_6/task2.9.js (hunk 4 above)
Comment (JackBlaaack) [reply JS_6/task2.9.js:23]: I see

File: JS_6/task2.3.js (hunk 2 above)
Comment (JackBlaaack) [reply JS_6/task2.3.js]: understood, I've fixed it

File: JS_6/task2.4.js (hunk 3 above)
Comment (JackBlaaack) [reply JS_6/task2.4.js]: I've fixed it

**Review by anatoly-karpovich — APPROVED** [review]

**Pull request merged**